pxr/imaging/hdx/shaders/renderPass.glslfx

```diff
@@ -176,6 +176,75 @@ vec4 IntToVec4(int id)
                 ((id >> 24) & 0xff) / 255.0);
 }
 
+// calculate an integer hash value based on integer key and offset
+// this is used for the pick buffer hash map
+int fnv1Hash(int key, int offset)
+{
+    const int prime = 16777619;
+
+    int ret = offset;
+	
+    int b0 = (key & 255);
+    int b1 = (key & 65280) >> 8;
+    int b2 = (key & 16711680) >> 16;
+    int b3 = (key & -2130706432) >> 24;
+
+    ret *= prime;
+    ret ^= b0;
+    ret *= prime;
+    ret ^= b1;
+    ret *= prime;
+    ret ^= b2;
+    ret *= prime;
+    ret ^= b3;
+
+    return ret;
+}
+
+bool 
+compareOrSet(
+    const int valueOffset, 
+    const int primId, 
+    const int instanceId, 
+    const int partId)
+{
+    int primIdOffset = valueOffset;
+    int instanceIdOffset = valueOffset + 1;
+    int partIdOffset = valueOffset + 2;
+
+    int primIdValue = PickBuffer[primIdOffset];
+    int instanceIdValue = PickBuffer[instanceIdOffset];
+    int partIdValue = PickBuffer[partIdOffset];
+
+    // if all the values are initialized, just compare 
+    if ((primIdValue != -9) && 
+        (instanceIdValue != -9) && 
+        (partIdValue != -9)) {
+        return  (primId == primIdValue) && 
+                (instanceId == instanceIdValue) && 
+                (partId == partIdValue);
+    }
+
+    // try to initialize the values and check if succeeded
+
+    primIdValue = atomicCompSwap(PickBuffer[primIdOffset], -9, primId);
```

Comment:
Here and elsewhere: this needs to be updated for metal support.  See top-of-tree WriteOitLayersToBufferCommon, in this same file, for the layout section and the ATOMIC_EXCHANGE macro usage.